### PR TITLE
[MOB-6648] update get messages to pull messages from placement

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s)

* [MOB-6648](https://iterable.atlassian.net/browse/MOB-6648)

## ✏️ Description

This pull request updates the syncMessages function to pull messages from the placement object in the payload.

[MOB-6648]: https://iterable.atlassian.net/browse/MOB-6648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by anlinguist** [comment]
1. When calling `getMessages(for placementId:)`, we need to make placementId a non-optional Int because server is returning placementId as a number type and without a placementId passed in, the alternative method `getMessages()` should be called.
2. We need to update tests.

Fixed in last commit.

**Review by Ayyanchira — APPROVED** [review]
LGTM. Nit pick comments left

File: swift-sdk/Internal/EmbeddedMessagingSerialization.swift [hunk 1]
```diff
@@ -72,10 +73,15 @@ struct EmbeddedMessagingSerialization {
     }
 }
 
-struct EmbeddedMessagesPayload: Codable {
+struct Placement: Codable {
+    let placementId: Int?
```
[the remark this answers is not in the document]
Comment (anlinguist) [reply swift-sdk/Internal/EmbeddedMessagingSerialization.swift:77]: The API is currently returning an Int, not a String, despite the Slab doc

**Review by anlinguist — APPROVED** [review]
lgtm

**Pull request merged**